packages/fxa-auth-server/lib/payments/stripe-formatter.ts

```diff
@@ -0,0 +1,45 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+import * as invoiceDTO from 'fxa-shared/dto/auth/payments/invoice';
+import { Stripe } from 'stripe';
+
+export function stripeInvoiceToInvoicePreviewDTO(
+  invoice: Stripe.Invoice
+): invoiceDTO.InvoicePreview {
+  const invoicePreview: invoiceDTO.invoicePreviewSchema = {
```

Comment:
> Is it possible to specify the response format with TS types instead of Joi to get rid of the redundancy and risk altogether? E.g. IIUC to update other routes, devs would have to remember to use both the DTO type and DTO schema in the approach you have here.

I considered that, but the typesafe-joi object type is not quite a normal interface, which is what the payments server works with. This does unfortunately rely on ensuring the implementation of this pattern returns the interface in one context, while the handler function itself has a return type of the Joi validation schema type, which ensures they structurally match.